src/main/java/com/github/havardh/javaflow/Execution.java

```diff
@@ -75,7 +78,7 @@ public Execution(
    * @return the flow type as a {@code String}
    */
   public String run(String... filenames) {
-    List<String> files = read(filenames);
```

Comment:
I'll just sum up this in a single comment. :)

This functionality makes a lot of sense, but I suggest a few changes for the following reasons. 
 1. reuse of this functionality between `javaflow`, `javaflow-maven-plugin` and any other plugins, this is already supported in `javaflow-maven-plugin`
 2. the `run` functionality reads fluently the abstraction layers in `javaflow` and I want to keep it (`read`, `parse`, `transform` `verify`, `write` and `transformFile`). 
 3. in contrast to the name `javaflow` is not tied to `java` or `flowtype` in design. So hardcoding `.java` in the main code is against my design goal. 

First of all we need to find a common ground between the functionality you propose and the current maven plugin. I propose the following format. `javaflow <file|directory|minimatch>...` where `file` is a single file, `directory` is like you suggest (read recursively) and `minimatch` is a [`minimatch`](https://github.com/isaacs/minimatch) pattern. The `javaflow-maven-plugin` can now concat the `suffix` onto the base dir as expected, we will just change the format of the `suffix` to a `minimatch` pattern. 

The user could run `javaflow /path/to/dir/**/*.java` to get the behaviour in this branch.

Next up, to keep the fluent `run` function I propose we add another layer before `read` called `resolve`. It should resolve a list of `<file|directory|minimatch>`'s into a list of the actual files to read. Then this list can be passed into the old `read` function.

It does of course not have to be the `minimatch` pattern we use. What do you think?

Reply:
Your point about supporting `file/directory/minimatch` sounds reasonable. The only question is how to determine if the argument is supposed to be a file or a minimatch pattern?

For Java I found `java.nio.file.FileSystem#getPathMatcher`, which supports some kind of glob, but not sure how/if it differs from minimatch.

Reply:
Glob is perfectly a great alternative, lets go with that. 

I suggest to check if the given parameter is a match on an actual file or directory and if not assume that it is a `glob` pattern and do a search. 

If we are to use `getPathMatcher` I guess we would have to figure out what the root directory of the search, list all files recursively and then use it to match against the `glob`. 
I used `apache-commons` `FileUtils.listFiles` in `javaflow-maven-plugin`. We could use it with [`WildcardFileFilter`](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/filefilter/WildcardFileFilter.html), but we would still have to figure out the root directory.